### PR TITLE
feat: add search endpoint

### PR DESCRIPTION
**Pull request by clostao**
Implements the backend side of https://github.com/autonomys/auto-drive/issues/2

### CODE REVIEW

**Comment by clostao** [comment]
@marc-aurele-besner @jfrank-summit I'm starting using PRs in this project as it starts to be more consolidated

File: backend/src/api/metadata.ts [hunk 1]
```diff
@@ -14,6 +14,13 @@ export const saveMetadata = async (cid: string, metadata: OffchainMetadata) => {
   return metadataRepository.setMetadata(cid, metadata);
 };
 
+export const searchMetadataByCID = async (cid: string, limit?: number) => {
```
Comment (marc-aurele-besner) [inline backend/src/api/metadata.ts:17]: Just trying to simplify
```suggestion
export const searchMetadataByCID = async (cid: string, limit: number =  5) => {
```

File: backend/src/api/metadata.ts [hunk 2]
```diff
@@ -14,6 +14,13 @@ export const saveMetadata = async (cid: string, metadata: OffchainMetadata) => {
...
+  const defaultLimit = 5;
+  return metadataRepository
+    .searchMetadataByCID(cid, limit ?? defaultLimit)
```
Comment (marc-aurele-besner) [inline backend/src/api/metadata.ts:20]:
```suggestion
    .searchMetadataByCID(cid, limit)
```

File: backend/src/api/metadata.ts [hunk 3]
```diff
@@ -14,6 +14,13 @@ export const saveMetadata = async (cid: string, metadata: OffchainMetadata) => {
   return metadataRepository.setMetadata(cid, metadata);
 };
 
+export const searchMetadataByCID = async (cid: string, limit?: number) => {
+  const defaultLimit = 5;
```
Comment (marc-aurele-besner) [inline backend/src/api/metadata.ts:18]:
```suggestion
```

**Review by marc-aurele-besner — APPROVED** [review]
LGTM

**Pull request merged**